packages/block-editor/src/components/editor-styles/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { compact, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import transformStyles from '../../utils/transform-styles';
+
+function EditorStyles( { styles } ) {
```

Comment:
Was there any consideration to make this a hook, considering its own behaviors are hooks? And observing that the current implementation forced refactoring of `EditorProvider` to render a fragment to support the additional top-level component.

Reply:
I thought about it, the main reason I didn't do it is because `block-editor` don't expose hooks but expose similar behavior components like "ObserveTyping", "WritingFlow". I guess even this behaviors could be hooks. 

Now, that I think about it, maybe a hook is better though.

Reply:
Yeah, I was thinking about these "behavioral" components as well. They made sense at the time. Nowadays, I feel that hooks are a perfect fit for how and why we were implementing those.

There's still an argument of "consistency for consistency's sake", so I could be okay with either approach.